ignite/services/plugin/template/main.go.plush

```diff
@@ -2,117 +2,45 @@ package main
-	}, nil
-}
-
-func (app) Execute(ctx context.Context, cmd *plugin.ExecutedCommand, api plugin.ClientAPI) error {
-	// TODO: write command execution here
-	fmt.Printf("Hello I'm the example-plugin plugin\n")
-	fmt.Printf("My executed command: %q\n", cmd.Path)
-	fmt.Printf("My args: %v\n", cmd.Args)
-
-	flags, err := cmd.NewFlags()
-	if err != nil {
-		return err
+func (app) Manifest(_ context.Context) (*plugin.Manifest, error) {
+	m := &plugin.Manifest{
+		Name: "<%= Name %>",<%= if (SharedHost) { %>
+		SharedHost: true,<% } %>
 	}
-
-	myFlag, _ := flags.GetString("my-flag")
-	fmt.Printf("My flags: my-flag=%q\n", myFlag)
-	fmt.Printf("My config parameters: %v\n", cmd.With)
-
-	// This is how the plugin can access the chain:
-	// c, err := getChain(cmd)
-	// if err != nil {
-	// 	return err
-	// }
-
-	// According to the number of declared commands, you may need a switch:
-	/*
-		switch cmd.Use {
-		case "add":
-			fmt.Println("Adding stuff...")
-		case "list":
-			fmt.Println("Listing stuff...")
-		case "delete":
-			fmt.Println("Deleting stuff...")
-		}
-	*/
-	
-	// ClientAPI call example
-	fmt.Println(api.GetChainInfo(ctx))
-	
-	return nil
+	m.ImportCobraCommand(cmd.New<%= Title %>(), "ignite")
+	return m, nil
 }
 
-func (app) ExecuteHookPre(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook pre %q\n", h.Hook.GetName())
-	return nil
+func (app) Execute(_ context.Context, c *plugin.ExecutedCommand, _ plugin.ClientAPI) error {
+	// Instead of a switch on c.Use, we run the root command like if
+	// we were in a command line context. This implies to set os.Args
+	// correctly.
+	// Remove the first arg "ignite" from OSArgs because our <%= Name %>
+	// command root is "<%= Name %>" not "ignite".
+	os.Args = c.OsArgs[1:]
+	return cmd.New<%= Title %>().Execute()
 }
 
-func (app) ExecuteHookPost(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook post %q\n", h.Hook.GetName())
+func (app) ExecuteHookPre(_ context.Context, _ *plugin.ExecutedHook, _ plugin.ClientAPI) error {
 	return nil
 }
 
-func (app) ExecuteHookCleanUp(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
-	fmt.Printf("Executing hook cleanup %q\n", h.Hook.GetName())
+func (app) ExecuteHookPost(_ context.Context, _ *plugin.ExecutedHook, _ plugin.ClientAPI) error {
 	return nil
 }
 
-func getChain(cmd *plugin.ExecutedCommand, chainOption ...chain.Option) (*chain.Chain, error) {
-	flags, err := cmd.NewFlags()
-	if err != nil {
-		return nil, err
-	}
-
-	var (
-		home, _ = flags.GetString("home")
-		path, _ = flags.GetString("path")
-	)
-	if home != "" {
-		chainOption = append(chainOption, chain.HomePath(home))
-	}
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return nil, err
-	}
-	return chain.New(absPath, chainOption...)
+func (app) ExecuteHookCleanUp(_ context.Context, _ *plugin.ExecutedHook, _ plugin.ClientAPI) error {
+	return nil
```

Comment:
Why was this removed? this a example for the user implementation

Reply:
Focus of this PR was to minimize the scaffolding code so that the end user doesn't have to remove any code. As for example of how hooks are used, we can have multiple sample apps in the `apps` repository.

Reply:
I'm not sure if `getChain()` is documented somewhere but it would be nice if an app example with that function is available to users in a tutorial or apps documentation because it could be a valuable reference.

Reply:
I will add this as an example in the `apps` repository.